dashboard/app/dsl/level_group_dsl.rb

```diff
@@ -5,11 +5,16 @@ def initialize
     @title = nil
     @description_short = nil
     @description = nil
-    @hash[:pages] = []
     @hash[:texts] = []
     @hash[:options] = {skip_dialog: true, skip_sound: true}
     @current_page_level_names = []
     @level_names = []
+    @pages = []
+  end
+
+  # @override
+  def parse_output
+    super.merge(pages: @pages)
```

Comment:
I found it useful to make a summary of the call stacks when I was implementing this. It may be helpful for reviewing too: https://docs.google.com/document/d/1ibP4ODlbD4e6VNl9FKQ_E3cMtyB-KNblf8UX59dNXTY/edit#heading=h.l6uwzoqw7v5s